esphome/components/tuya/climate/__init__.py

```diff
@@ -99,27 +153,25 @@ def validate_eco_values(value):
             cv.Optional(CONF_SUPPORTS_HEAT, default=True): cv.boolean,
             cv.Optional(CONF_SUPPORTS_COOL, default=False): cv.boolean,
             cv.Optional(CONF_SWITCH_DATAPOINT): cv.uint8_t,
-            cv.Optional(CONF_ACTIVE_STATE_DATAPOINT): cv.uint8_t,
-            cv.Optional(CONF_ACTIVE_STATE_HEATING_VALUE, default=1): cv.uint8_t,
-            cv.Optional(CONF_ACTIVE_STATE_COOLING_VALUE): cv.uint8_t,
+            cv.Optional(CONF_ACTIVE_STATE): ACTIVE_STATES,
```

Comment:
Don't apply this suggestion as it requires more changes, but for removing config options, we switch them to `cv.invalid` do people get nicer errors when they try to compile and remove them in a later version.

```suggestion
            cv.Optional(CONF_ACTIVE_STATE): ACTIVE_STATES,
            cv.Optional("active_state_datapoint"): cv.invalid("'active_state_datapoint' has moved inside of the 'active_state' config block as 'datapoint'"),
            cv.Optional("active_state_heating_value"): cv.invalid(...),
            cv.Optional("active_state_cooling_value"): cv.invalid(...),
```